src/drivers/mediatek/mt8186/ipc.c

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Mediatek. All rights reserved.
+//
+// Author: Allen-KH Cheng <allen-kh.cheng@mediatek.com>
+//         Tinghan Shen <tinghan.shen@mediatek.com>
+
+#include <sof/debug/panic.h>
+#include <sof/drivers/interrupt.h>
+#include <sof/ipc/driver.h>
+#include <sof/ipc/msg.h>
+#include <sof/ipc/schedule.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/dma.h>
+#include <sof/lib/mailbox.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/uuid.h>
+#include <sof/lib/wait.h>
+#include <sof/list.h>
+#include <sof/platform.h>
+#include <sof/schedule/edf_schedule.h>
+#include <sof/schedule/schedule.h>
+#include <sof/schedule/task.h>
+#include <sof/spinlock.h>
+#include <ipc/header.h>
+#include <ipc/topology.h>
+#include <ipc/trace.h>
+#include <platform/drivers/mt_reg_base.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#define IPC_DSPMBOX_DSP_RSP 0
+#define IPC_DSPMBOX_DSP_REQ 1
+
+/* 389c9186-5a7d-4ad1-a02c-a02ecdadfb33 */
+DECLARE_SOF_UUID("ipc-task", ipc_task_uuid, 0x389c9186, 0x5a7d, 0x4ad1,
```

Comment:
Need to generate another unique uuid for mt8186, don't use the same one with mt8195.